src/main/webapp/shopping-results.js

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let photoCategory;
+let blobKeyString;
+
+/**  
+ * Gets and stores information about the photo uploaded by the user, by making a GET request
+ * to '/get-image-info'. 
+ * {@code photoCategory} and {@code blobKeyString} are needed for making the GET request to 
+ * the PhotoShoppingServlet.
+ */
+async function fetchUploadedImageInfo() {
+  const response = await fetch('/get-image-info')
+      .catch((error) => {
+        console.warn(error);
+        return new Response(JSON.stringify({
+          code: 400,
```

Comment:
Why is this 400? Shouldn't this be decided by the response instead of a fix code here?

Reply:
Right, I changed it now!